### PR TITLE
Log AuthnContextClassRef on successful login

### PR DESCRIPTION
**Pull request by pablothedude**
The Loa was not logged when AM was being used. This change
logs the authncontextclassref instead which won't get lost during
deserializing the entiy when using AM.

https://www.pivotaltracker.com/story/show/171404731

### CODE REVIEW

File: library/EngineBlock/Corto/Filter/Command/LogLogin.php [hunk 1]
```diff
@@ -48,16 +48,14 @@ public function execute()
...
-
         $this->authenticationLogger->logLogin(
             $this->_serviceProvider,
             $this->_identityProvider,
             $this->_collabPersonId,
             $this->_request->getKeyId(),
             $requesterChain,
-            $originalResponse->getIntendedNameId(),
-            $originalResponse->getAttainedLoa()
+            $this->_response->getNameIdValue(),
```
Comment (MKodde) [inline library/EngineBlock/Corto/Filter/Command/LogLogin.php:57]: The intended name id seems to be set with the collab person id. Where the name id could have been changed to another value in the AM or other output filters.

- Why did you change behavior regarding logging the name id?
- Was this part of the ticket? (please link the ticket)

File: library/EngineBlock/Corto/Module/Service/ProcessedAssertionConsumer.php [hunk 2]
```diff
@@ -55,7 +55,6 @@ public function serve($serviceName, Request $httpRequest)
             $this->_server->setKeyId($receivedRequest->getKeyId());
         }
 
-        $currentProcess = $this->_processingStateHelper->getStepByRequestId($receivedRequest->getId(), ProcessingStateHelperInterface::STEP_CONSENT);
```
Comment (MKodde) [inline library/EngineBlock/Corto/Module/Service/ProcessedAssertionConsumer.php:58]: This seems unrelated to the context of this commit. However, removing this unused code is a big :+1: maybe put it on a dedicated commit?

File: src/OpenConext/EngineBlock/Logger/AuthenticationLogger.php [hunk 3]
```diff
@@ -82,7 +82,7 @@ function (Entity $entity) {
                 'proxied_sp_entity_ids' => $proxiedServiceProviderEntityIds,
                 'workflow_state'        => $workflowState,
                 'original_name_id'      => $originalNameId,
-                'attained_loa'          => $attainedLoa,
+                'authncontextclassref'  => $authnContextClassRef,
```
Comment (MKodde) [inline src/OpenConext/EngineBlock/Logger/AuthenticationLogger.php:85]: Naming typo:
~authncontextclassref~ `authn_context_class_ref`

Comment (pablothedude) [reply src/OpenConext/EngineBlock/Logger/AuthenticationLogger.php:85]: It's the actual name of the attribute: `AuthnContextClassRef` so it's one word. I've also validated it with the PM so I leave it as-is.

File: library/EngineBlock/Corto/Module/Service/ProcessedAssertionConsumer.php (hunk 2 above)
Comment (pablothedude) [reply library/EngineBlock/Corto/Module/Service/ProcessedAssertionConsumer.php:58]: I reckoned it wouldn't be that valuable to document unused code.

File: src/OpenConext/EngineBlock/Logger/AuthenticationLogger.php (hunk 3 above)
Comment (MKodde) [reply src/OpenConext/EngineBlock/Logger/AuthenticationLogger.php:85]: That makes no sense to me, but I don't mind leaving it like this either.. Maybe for consistency sake also rename the variable used in the assignement to `$authncontextclassref`?

File: library/EngineBlock/Corto/Module/Service/ProcessedAssertionConsumer.php (hunk 2 above)
Comment (MKodde) [reply library/EngineBlock/Corto/Module/Service/ProcessedAssertionConsumer.php:58]: It makes the commit less atomic. Personally I'd put it in a commit of it's own but feel free to leave it here.

File: library/EngineBlock/Corto/Filter/Command/LogLogin.php (hunk 1 above)
Comment (pablothedude) [reply library/EngineBlock/Corto/Filter/Command/LogLogin.php:57]: https://www.pivotaltracker.com/story/show/171404731/comments/214302244

**Comment by pablothedude** [comment]
No changes were made to the branch, only the PR itself has been changed.

File: src/OpenConext/EngineBlock/Logger/AuthenticationLogger.php (hunk 3 above)
Comment (thijskh) [reply src/OpenConext/EngineBlock/Logger/AuthenticationLogger.php:85]: For the record I have no particular attachment to the specific key name (whether it contains `_` or not I mean). The saml tag name is indeed withput _ but I see the point with matching the underscores of the other keys in this array. In any case, I guess both would be fine choices.

File: library/EngineBlock/Corto/Filter/Command/LogLogin.php (hunk 1 above)
Comment (MKodde) [reply library/EngineBlock/Corto/Filter/Command/LogLogin.php:57]: Thanks, that info was missing

File: src/OpenConext/EngineBlock/Logger/AuthenticationLogger.php (hunk 3 above)
Comment (MKodde) [reply src/OpenConext/EngineBlock/Logger/AuthenticationLogger.php:85]: From a consistency and coding standard standpoint I'd say use the underscores. But I agree, that from a functional standpoint this does not add much/any value.

**Comment by pablothedude** [comment]
Approved in Slack
